### PR TITLE
More flexible integration test

### PR DESCRIPTION
**Pull request by SpacePossum**
replaces https://github.com/twigphp/Twig/pull/1805 as a 1.x version

### CODE REVIEW

File: lib/Twig/Test/IntegrationTestCase.php [hunk 1]
```diff
@@ -17,10 +17,44 @@
...
+     */
+    protected function getFilters()
+    {
+        return array();
+    }
+
+    /**
+     * @return Twig_SimpleFunction[]
+     */
+    protected function getFunctions()
```
Comment (Tobion) [inline lib/Twig/Test/IntegrationTestCase.php]: how about getTwigFunctions to be consistent?

File: test/Twig/Tests/LegacyIntegrationTest.php [hunk 2]
```diff
@@ -22,6 +22,15 @@ public function getFixturesDir()
...
     }
+
+    public function getTests($name, $legacyTests = false)
+    {
+        if (!$legacyTests) {
+            return array(array('not', '-', '', array(), '', array()));
+        }
+
+        return parent::getTests($name, true);
+    }
```
Comment (SpacePossum) [inline test/Twig/Tests/LegacyIntegrationTest.php]: The test will no fail again if there are no fixtures to test (a minor BC which is reverted by this). 
The exception on that rule is now placed in the test that needs that exception and no longer in the parent.

File: lib/Twig/Test/IntegrationTestCase.php [hunk 3]
```diff
@@ -17,10 +17,44 @@
...
-    abstract protected function getExtensions();
+    /**
+     * @return string
+     */
     abstract protected function getFixturesDir();
 
     /**
+     * @return Twig_ExtensionInterface[]
+     */
+    protected function getExtensions()
```
Comment (SpacePossum) [inline lib/Twig/Test/IntegrationTestCase.php]: this cannot be renamed since it would be a BC break

File: lib/Twig/Test/IntegrationTestCase.php [hunk 4]
```diff
@@ -17,10 +17,44 @@
...
+    {
+        return array();
+    }
+
+    /**
+     * @return Twig_SimpleFilter[]
+     */
+    protected function getTwigFilters()
```
Comment (fabpot) [inline lib/Twig/Test/IntegrationTestCase.php]: Why not just `getFilters`?

**Comment by fabpot** [comment]
Do you any use case in mind? I don't see when this could be interesting for end users.

File: lib/Twig/Test/IntegrationTestCase.php (hunk 4 above)
Comment (Tobion) [inline lib/Twig/Test/IntegrationTestCase.php]: because `getTests` is already taken and thus it would be inconsistent

**Comment by SpacePossum** [comment]
Yes :)
I use this in my projects to test custom `filters`, so I create a `.test` file, set the fixture directory and I'm all set. That is why I added the option to just register custom filters and not only `extensions` (and allow not setting `extensions`).

The second thing I try to fix here; is that it used to be that this test would fail if it won't find any `.test` files, but now the test passes (even for non-legacy items).

So yes, I'm trying to fix two things in one PR once more, so I'm happy to revert/split up whatever needed.

File: lib/Twig/Test/IntegrationTestCase.php (hunk 4 above)
Comment (SpacePossum) [inline lib/Twig/Test/IntegrationTestCase.php]: I'm happy to name it whatever you (Stof, Tobian and Fabien) think is best

**Comment by fabpot** [comment]
Thank you @SpacePossum.

**Comment by SpacePossum** [comment]
thank you @fabpot :+1:

**Pull request closed without merging**